include/manifold/linalg.h

```diff
@@ -0,0 +1,2189 @@
+  return fold(detail::max{}, a.x, a);
+}
+template <class T, int M>
+int argmin(const vec<T, M> &a) {
+  int j = 0;
+  for (int i = 1; i < M; ++i)
+    if (a[i] < a[j]) j = i;
+  return j;
+}
+template <class T, int M>
+int argmax(const vec<T, M> &a) {
+  int j = 0;
+  for (int i = 1; i < M; ++i)
+    if (a[i] > a[j]) j = i;
+  return j;
+}
+
+// Unary operators are defined component-wise for linalg types
+template <class A>
+constexpr apply_t<detail::op_pos, A> operator+(const A &a) {
+  return apply(detail::op_pos{}, a);
+}
+template <class A>
+constexpr apply_t<detail::op_neg, A> operator-(const A &a) {
+  return apply(detail::op_neg{}, a);
+}
+template <class A>
+constexpr apply_t<detail::op_cmp, A> operator~(const A &a) {
+  return apply(detail::op_cmp{}, a);
+}
+template <class A>
+constexpr apply_t<detail::op_not, A> operator!(const A &a) {
+  return apply(detail::op_not{}, a);
+}
+
+// Binary operators are defined component-wise for linalg types, EXCEPT for
+// `operator *`
+template <class A, class B>
+constexpr apply_t<detail::op_add, A, B> operator+(const A &a, const B &b) {
+  return apply(detail::op_add{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_sub, A, B> operator-(const A &a, const B &b) {
+  return apply(detail::op_sub{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_mul, A, B> cmul(const A &a, const B &b) {
+  return apply(detail::op_mul{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_div, A, B> operator/(const A &a, const B &b) {
+  return apply(detail::op_div{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_mod, A, B> operator%(const A &a, const B &b) {
+  return apply(detail::op_mod{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_un, A, B> operator|(const A &a, const B &b) {
+  return apply(detail::op_un{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_xor, A, B> operator^(const A &a, const B &b) {
+  return apply(detail::op_xor{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_int, A, B> operator&(const A &a, const B &b) {
+  return apply(detail::op_int{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_lsh, A, B> operator<<(const A &a, const B &b) {
+  return apply(detail::op_lsh{}, a, b);
+}
+template <class A, class B>
+constexpr apply_t<detail::op_rsh, A, B> operator>>(const A &a, const B &b) {
+  return apply(detail::op_rsh{}, a, b);
+}
+
+// Binary `operator *` was originally defined component-wise for all patterns,
+// in a fashion consistent with the other operators. However, this was one of
+// the most frequent sources of confusion among new users of this library, with
+// the binary `operator *` being used accidentally by users who INTENDED the
+// semantics of the algebraic matrix product, but RECEIVED the semantics of the
+// Hadamard product. While there is precedent within the HLSL, Fortran, R, APL,
```

Comment:
This is dumb - I hate Hadamard products. I'd like to switch to the opposite syntax: make `*` an alias for `mul`. @pca006132 I'm having a little trouble following their templating syntax - do you know how to make this change?

Reply:
Okay, I think I figured this one out.